File: Days 001 - 010/Day 01/ListContainingSumOfTwoNumbers.cpp
#include <iostream>
#include <unordered_set>
#include <vector>

bool ContainsTwoNumberSum(const std::vector<int>& numberList, int total)
{
	std::unordered_set<int> numberHash;

	for (unsigned int i = 0; i < numberList.size(); i++)
	{
		if (numberHash.find(total - numberList[i]) != numberHash.end())
		{
			return true;
		}
		
		numberHash.insert(numberList[i]);
	}
	
	return false;
}

int main(int argc, char* argv[])
{
	std::vector<int> numbers { 10, 15, 3, 7 };

	std::cout << std::boolalpha;
	std::cout << ContainsTwoNumberSum(numbers, 17) << "\n";

	std::cin.get();

	return 0;
}

File: Days 001 - 010/Day 01/ListContainingSumOfTwoNumbers.lua
function contains_two_number_sum(array, total)
	local number_hash = { }

	for i = 1, #array
	do
		if number_hash[total - array[i]] ~= nil
		then
			return true
		end

		number_hash[array[i]] = true
	end

	return false
end

io.write(tostring(contains_two_number_sum({ 10, 15, 3, 7 }, 17)), "\n")